src/github/mod.rs

```diff
@@ -182,6 +186,241 @@ impl SyncGitHub {
         }))
     }
 
+    fn diff_repos(&self) -> Result<Vec<RepoDiff>, Error> {
+        let mut diffs = Vec::new();
+        for repo in &self.repos {
+            diffs.push(self.diff_repo(repo)?);
+        }
+        Ok(diffs)
+    }
+
+    fn diff_repo(&self, expected_repo: &rust_team_data::v1::Repo) -> Result<RepoDiff, Error> {
+        let actual_repo = match self.github.repo(&expected_repo.org, &expected_repo.name)? {
+            Some(r) => r,
+            None => {
+                let mut permissions = Vec::new();
+                for expected_team in &expected_repo.teams {
+                    permissions.push((
+                        RepoCollaborator::Team(expected_team.name.clone()),
+                        convert_permission(&expected_team.permission),
+                    ));
+                }
+
+                for bot in &expected_repo.bots {
+                    permissions.push((
+                        RepoCollaborator::User(bot_name(bot).to_owned()),
+                        RepoPermission::Write,
+                    ));
+                }
+
+                for member in &expected_repo.members {
+                    permissions.push((
+                        RepoCollaborator::User(member.name.clone()),
+                        convert_permission(&member.permission),
+                    ));
+                }
+
+                let mut branch_protections = Vec::new();
+                for branch in &expected_repo.branches {
+                    branch_protections.push((
+                        branch.name.clone(),
+                        branch_protection(expected_repo, branch),
+                    ));
+                }
+
+                return Ok(RepoDiff::Create(CreateRepoDiff {
+                    org: expected_repo.org.clone(),
+                    name: expected_repo.name.clone(),
+                    description: expected_repo.description.clone(),
+                    permissions,
+                    branch_protections,
+                }));
+            }
+        };
+
+        let permission_diffs = self.diff_permissions(expected_repo)?;
+        let branch_protection_diffs = self.diff_branch_protections(&actual_repo, expected_repo)?;
+        let name_diff =
+            (actual_repo.name != expected_repo.name).then(|| expected_repo.name.clone());
```

Comment:
What change is this supposed to catch?

Reply:
Repos can be renamed, but I suppose since we're using the name as the id, this diff would never actually trigger. I can remove it.